lib/Models/Leaflet.js

```diff
@@ -415,42 +409,51 @@ function createLeafletCredit(attribution) {
     return element.outerHTML;
 }
 
-function featurePicked(leaflet, entity) {
+/*
+* Feature picking in leaflet is funny because we rely on multiple click events for picking vector features and a single
+* map click event for picking raster features. These can come in in any order, so we handle it like this
+*
+* - Initially, leaflet._pickedFeatures is undefined
+* - When the first click event is triggered (whether is a map or feature click), it starts the asynchronous raster
+*       feature picking, sets terria.pickedFeatures and also spawns a runLater that will clear _pickedFeatures
+* - Each feature pick, including the first, adds a feature to _pickedFeatures.features.
+* - The map click is basically ignored, unless it is the first click event we see.
+* - Once this runLater executes, we can assume that all the click handlers for that one click will have fired, so we
+*       set _pickedFeatures back to undefined (this.terria.pickedFeatures is still set)
+* - When another click happens, it re-initialized _pickedFeatures and the process starts again.
+*
+* For most types of features, the map click will be the first click event, which means the raster pick will happen
+* exactly where the user clicked.  Leaflet billboards, however, seem to swallow click events, so if the user clicks
+* on a billboard _only_ the click for that feature will be raised.  That means the raster pick will happen at the
+* location of the billboard, rather than at the actual location that the user clicked.
+*/
```

Comment:
I know this largely pre-dates your PR, but this whole comment block is really confusing (to me at least). "Basically" ignored? What if the map click _is_ the first we see? Do we really rely on multiple clicks for vector features, when we just use one of them? (Maybe "rely on" should be "receive"?) If "multiple clicks" is always two, why not just say two? Etc.

I propose this replacement:

```
/*
* There are two "listeners" for clicks which are set up in our constructor.
* - One fires for any click: `map.on('click', ...`.  It calls `pickFeatures`.
* - One fires only for vector features: `this.scene.featureClicked.addEventListener`. It calls `featurePicked`, which calls `pickFeatures` and then adds the feature it found, if any.
* These events can fire in either order.
* Billboards do not fire the first event.
*
* Note that `pickFeatures` does nothing if `leaflet._pickedFeatures` is already set. 
* Otherwise, it sets it, runs `runLater` to clear it, and starts the asynchronous raster feature picking.
*
* So:
* If only the first event is received, it triggers the raster-feature picking as desired.
* If both are received in the order above, the second adds the vector features to the list of raster features as desired.
* If both are received in the reverse order, the vector-feature click kicks off the same behavior as the other click would have;
* and when the next click is received, it is ignored - again, as desired.
*/
```